web/pandas/pdeps/0008-inplace-methods-in-pandas.md

```diff
@@ -0,0 +1,403 @@
+# PDEP-8: In-place methods in pandas
+
+- Created: 16 February 2023
+- Status: Under discussion
+- Discussion: [PR xxx](https://github.com/pandas-dev/pandas/pull/xxx)
+- Authors: [Thomas Li](https://github.com/lithomas1),
+           [Patrick Hoefler](https://github.com/phofl),
+           [Joris Van den Bossche](https://github.com/jorisvandenbossche)
+- Revision: 1
+
+## Abstract
+
+This PDEP proposes that:
+
+- The ``inplace`` parameter will be removed from any methods that never can be done inplace
+- The ``inplace`` parameter will also be removed from any methods that modify the shape of a pandas object's values or
+  don't modify the internal values of a pandas object at all.
+- In contrast, the ``inplace`` parameter will be kept for any methods that only modify the underlying data of a pandas
+  object.
+    - For example, the ``fillna`` method would retain its ``inplace`` keyword, while ``dropna`` (potentially shrinks the
+      length of a ``DataFrame``/``Series``) and ``rename`` (alters labels not values) would lose their ``inplace``
+      keywords
+    - For those methods, since Copy-on-Write behavior will lazily copy if the result is unchanged, users should reassign
+      to the same variable to imitate behavior of the ``inplace`` keyword.
+      e.g. ``df = df.dropna()`` for a DataFrame with no null values.
+- The ``copy`` parameter will also be removed, except in constructors and in functions/methods that convert array-likes
+  to pandas objects (e.g. the ``pandas.array`` function) and functions/methods that export pandas objects to other data
+  types (e.g. ``DataFrame/Series.to_numpy`` method).
+- Open Questions
+  (These questions are deferred to a later revision, and will not affect the acceptance process of this PDEP.)
+    - Should ``inplace=True`` return the original pandas object that was operated inplace on?
+    - What should happen when ``inplace=True`` but the original pandas object cannot be operated inplace on because it
+      shares its values with another object?
```

Comment:
This is a good point, nested data are not covered by CoW right now, it's only about arrays and pandas objects. An example would be the following:

```
ser = Series([1, 2, 3])
view = ser[:]
# Both share the same data since [:] creates a view
ser.replace(1, 5, inplace=True)  # makes a copy of the data because we don't want to update view
```